### DISCUSSION TITLE
Hosting on Windows Server 2022

### DISCUSSION THREAD

**Discussion by abdulsalamhq** [heyform/heyform#97] [Self hosting]
I've got a Windows server 2022 where I'm deploying Heyform. Here are the steps I'm following:

1. Clone the project.
2. In the root folder, run the command "pnpm install" to add node_modules in the server and webapp.
3. Create a build. Unfortunately, "pnpm build:deps" didn't work for me, so I had to create the build separately by running "pnpm build" in the server and webapp.
4. Create a project in IIS and point it to the webapp.
5. When I run the webapp (Front-End), it works fine. However, when I click on the login button, it gives me a 404 error.

UI works fine but when I log in it gives me an error 404 for the API

**Comment by iMuFeng** [comment]
Can you share your config or the login API http request log here?

**Reply by abdulsalamhq** [reply]
<configuration>
   <system.webServer>

     

     <handlers>
       <add name="iisnode" path="main.js" verb="*" modules="iisnode" />
     </handlers>

     

     <rewrite>
       <rules>
         <rule name="rewriteRule">
           <match url="/*" />
           <action type="Rewrite" url="main.js" />
         </rule>
       </rules>
     </rewrite>

     
     
     <security>
       <requestFiltering>
         <hiddenSegments>
           <add segment="node_modules" />
         </hiddenSegments>
       </requestFiltering>
     </security>    
     
   </system.webServer>
 </configuration>

**Reply by iMuFeng** [reply]
The project workflow involves packaging the `webapp` into HTML, JS, and CSS files. These packaged files are then copied to specific folders in the `server` as indicated in this [mkdir -p ../server/static && cp -r dist/static/* ../server/static/ && cp dist/index.html ../server/view/](https://github.com/heyform/heyform/blob/b253799c57885ed2cb078bd5187eba01203ae842/packages/webapp/package.json#L13). 

Finally, you just need to start the `server`.

**Reply by abdulsalamhq** [reply]
can you elaborate on how to copy these or should do manually copy also this command is not work for me pnpm build:deps

**Reply by abdulsalamhq** [reply]
what you mean by packaging ?

**Reply by abdulsalamhq** [reply]
![Screenshot_21-5-2024_145555_localhost](https://github.com/heyform/heyform/assets/38523330/50562600-38d1-4c19-ad85-a4c95a841ec1)

**Reply by abdulsalamhq** [reply]
@iMuFeng I solved this issue am going to closed this issue.
